smithy-diff/src/main/java/software/amazon/smithy/diff/evaluators/AddedOperationError.java

```diff
@@ -42,15 +43,23 @@ private List<ValidationEvent> createErrorViolations(ChangedShape<OperationShape>
         }
 
         List<ValidationEvent> events = new ArrayList<>();
-        for (ShapeId id : change.getNewShape().getErrors()) {
-            if (!change.getOldShape().getErrors().contains(id)) {
-                events.add(warning(change.getNewShape(), String.format(
-                        "The `%s` error was added to the `%s` operation. This "
-                        + "is backward-compatible if the error is only "
-                        + "encountered as a result of a change in behavior of "
-                        + "the client (for example, the client sends a new "
-                        + "parameter to an operation).",
-                        id, change.getShapeId())));
+        for (ShapeId error : change.getNewShape().getErrors()) {
+            if (!change.getOldShape().getErrors().contains(error)) {
+                events.add(
+                        ValidationEvent.builder()
+                                .id(getEventId() + "." + error)
+                                .severity(Severity.WARNING)
+                                .message(String.format(
+                                        "The `%s` error was added to the `%s` operation. This "
+                                                + "is backward-compatible if the error is only "
+                                                + "encountered as a result of a change in behavior of "
+                                                + "the client (for example, the client sends a new "
+                                                + "parameter to an operation).",
+                                        error, change.getShapeId()))
+                                .shape(change.getNewShape())
+                                .sourceLocation(change.getNewShape().getSourceLocation())
```

Comment:
nit - no need for this when using `.shape`

Reply:
ack